app/models/user.rb

```diff
@@ -21,6 +21,7 @@ class User < ApplicationRecord
   validate :referral_exists
   validates :data, data_is_json: true
   validates :email,       email: true, presence: true, uniqueness: true
+  validates :nickname,    length: { minimum: 4, maximum: 12 }, format: { with: /\A[a-z0-9]+\z/ }, uniqueness: true
```

Comment:
we should probably force to lowercase somewhere

Reply:
I think the `format: { with: /\A[a-z0-9]+\z/ }` is already validate for accept only lowercase. 
Or... Do you think we should allow `format: { with: /\A[a-zA-Z0-9]+\z/ }` and make somthing like?:

```ruby
before_save :downcase_fields

def downcase_fields
  self.nickname.downcase!
end
```